psc-flink/src/main/java/com/pinterest/flink/streaming/connectors/psc/internals/PscFetcher.java

```diff
@@ -135,15 +136,22 @@ public void runFetchLoop() throws Exception {
             while (running) {
                 // this blocks until we get the next records
                 // it automatically re-throws exceptions encountered in the consumer thread
-                final PscConsumerPollMessageIterator<byte[], byte[]> records = handover.pollNext();
 
-                // get the records for each topic partition
-                for (PscTopicUriPartitionState<T, TopicUriPartition> partition : subscribedPartitionStates()) {
-
-                    Iterator<PscConsumerMessage<byte[], byte[]>> partitionRecords =
-                            records.iteratorFor(partition.getPscTopicUriPartitionHandle());
-
-                    topicUriPartitionConsumerRecordsHandler(partitionRecords, partition);
+                try (PscConsumerPollMessageIterator<byte[], byte[]> records = handover.pollNext()) {
+                    while (records.hasNext()) {
+                        PscConsumerMessage<byte[], byte[]> record = records.next();
+                        PscTopicUriPartition
+                            key =
+                            new PscTopicUriPartition(
+                                record.getMessageId().getTopicUriPartition().getTopicUri(),
+                                record.getMessageId().getTopicUriPartition().getPartition());
```

Comment:
Aren't we creating too many objects here? Can't we reuse them?

Reply:
Fixed, thanks for noticing it!